physical/raft/raft_autopilot.go

```diff
@@ -389,21 +389,31 @@ func (d *Delegate) KnownServers() map[raft.ServerID]*autopilot.Server {
 // goroutine.
 func (d *Delegate) RemoveFailedServer(server *autopilot.Server) {
 	go func() {
-		if _, ok := d.deadServerRemovals.Load(server.ID); ok {
+		success := false
+		d.l.Lock()
```

Comment:
Why not add a new mutex to Delegate instead?  It gets harder to reason about a lock the more things it's used for and the more different goroutines use it.

Reply:
Regardless of whether you use this mutex or another, I'd be inclined to only hold the luck while accessing/mutating the map.  I don't know how raft.RemoveServer works, so it might not make a difference, but I don't see any reason to serialize parallel removal attempts of different servers.

Reply:
With each invocation of this method, we access the map. So we need to hold the lock each time.

This now returns early if the removal for a server is already triggered. If it is not triggered, it will initiate one in parallel and block on that.

Reply:
I don't think we need to hold the lock while we're waiting for RemoveServer, but I also don't want to hold up this PR on that point. It only really matters if RemoveServer could get stuck on serverA but work fine on a subsequent call with serverB, which seems unlikely.

Reply:
I appreciate the thoroughness. You are not holding up this PR. I want to get this right. Pushed up. Let me know how it looks.